packages/expression-parser/src/expression-parser/ExpressionParser.js

```diff
@@ -56,6 +61,7 @@ export class ExpressionParser {
         );
       }
     });
+    this.validExpressionCache.add(expression);
```

Comment:
I believe always caching at this point should be safe, as we ought to throw and exception already if the expression is invalid